util/codec/codec.go

```diff
@@ -984,10 +984,19 @@ func (decoder *Decoder) DecodeOne(b []byte, colIdx int, ft *types.FieldType) (re
 		chk.AppendBytes(colIdx, v)
 	case decimalFlag:
 		var dec *types.MyDecimal
-		b, dec, _, _, err = DecodeDecimal(b)
+		var frac int
+		b, dec, _, frac, err = DecodeDecimal(b)
 		if err != nil {
 			return nil, errors.Trace(err)
 		}
+		if ft.Decimal != types.UnspecifiedLength && frac > ft.Decimal {
```

Comment:
Is this check covered by any test cases?

Reply:
Yes, this BUG is triggered by test cases after opening CAST swither.